### ISSUE TITLE
Generated serialization code does not build when using custom nullable properties

### ISSUE THREAD

**Issue by joseharriaga** [Azure/autorest.csharp#3474]
Consider a property that was customized only to add a doc comment like this:

```csharp
/// Custom doc comment.
public int? MyProperty { get; set; }
```
 
OBSERVED (20230604.1)
The generated serialization code for this property using version 20230604.1 does not build due to error "cannot convert from 'int?' to 'decimal'":

```csharp
if (Optional.IsDefined(MyProperty))
{
    writer.WritePropertyName("my_property"u8);
    writer.WriteNumberValue(MyProperty);
}
```

EXPECTED
Notice the use of `MyProperty.Value` instead of `MyProperty`:

```csharp
if (Optional.IsDefined(MaxTokens))
{
    writer.WritePropertyName("my_property"u8);
    writer.WriteNumberValue(MyProperty.Value);
}
```

For the record, I also tried version 20230523.1 to try to understand if this was a regression. The generated serialization in that case was different, but also incorrect.

OBSERVED (20230523.1)
```csharp
if (Optional.IsDefined(MyProperty))
{
    writer.WritePropertyName("my_property"u8);
    writer.WriteNumberValue(MyProperty.Value.Value);
}
```

💡 Interestingly, if I remove the custom property, the serialization code is generated correctly. This is despite the fact that the generated property is equal to the custom one except only for the doc comment.

**Comment by ArcturusZhang** [comment]
I did some experiments, and I could reproduce this issue.
This issue only happens on DPG, mgmt/HLC works fine.
I will open a fix for this very soon

**Comment by ArcturusZhang** [comment]
Hi @joseharriaga this issue has been fixed in 0607.1 version of the generator.
